MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.Api.Tests/Controllers/ListControllerTests.cs

```diff
@@ -0,0 +1,134 @@
+{
+    [TestFixture]
+    internal class ListControllerTests
+    {
+        private ListController _listController;
+        private readonly ListItemEqualityComparer _listItemEqualityComparer = new ListItemEqualityComparer();
+
+        private readonly ListItem[] _items =
+        {
+            new ListItem
+            {
+                Id = new Guid("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAC"),
+                Text = "aaaaa",
+                IsActive = false,
+                CreationTime = new DateTime(1589, 12, 3),
+                LastUpdateTime = new DateTime(1896, 4, 7)
+            },
+            new ListItem
+            {
+                Id = new Guid("11AC59B7-9517-4EDD-9DDD-EB418A7C1644"),
+                Text = "dfads",
+                IsActive = false,
+                CreationTime = new DateTime(4568, 6, 23),
+                LastUpdateTime = new DateTime(8569, 8, 24)
+            },
+        };
+
+        private IListRepository _repository;
+        private IUrlLocator _location;
+
+        [SetUp]
+        public void Init()
+        {
+            _repository = Substitute.For<IListRepository>();
+            _location = Substitute.For<IUrlLocator>();
+
+            _listController = new ListController(_repository, _location)
+            {
+                Request = new HttpRequestMessage(),
+                Configuration = new HttpConfiguration()
+            };
+        }
+
+        [Test]
+        public async Task Get_ReturnsListOfItems()
+        {
+            _repository.GetAllItemsAsync().Returns(Task.FromResult(_items));
+
+            var message = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message.TryGetContentValue(out IEnumerable<ListItem> items);
+
+            Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(items, Is.EqualTo(_items).Using(_listItemEqualityComparer));
+        }
+
+        [Test]
+        public async Task Get_ReturnsItemWithGivenId()
+        {
+            var expectedItem = _items[0];
+            _repository.GetItemAsync(expectedItem.Id).Returns(Task.FromResult(expectedItem));
+
+            var message = await _listController.ExecuteAction(controller => controller.GetAsync(expectedItem.Id));
+            message.TryGetContentValue(out ListItem item);
+
+            Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(item, Is.EqualTo(expectedItem).Using(_listItemEqualityComparer));
+        }
+
+        [Test]
+        public async Task Post_CreatedReturned()
+        {
+            var newItem = new ListItem { Text = "newItem" };
+            var createdItem = new ListItem
+            {
+                Id = new Guid("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAB"),
+                Text = "newItem",
+                IsActive = false,
+                CreationTime = new DateTime(1589, 12, 3),
+                LastUpdateTime = new DateTime(1896, 4, 7)
```

Comment:
I love the dates 🤣 I can see that completing some items may sometimes take much longer than others 😆